### ISSUE TITLE
Go to Definition not working as expected for JSX tag

### ISSUE THREAD

**Issue by patrickgaskill** [Microsoft/vscode#28137]
- VSCode Version: 1.12.2
- OS Version: Windows 10 Pro 10.0.14393 N/A Build 14393

I've tried to isolate the issue to the following example, which is saved in `goto-test.js`.
```
import React from "react";
import Spinner from "components/spinner";
import { myConstants } from "constants/my-constants";

function GoToTest(props) {
  const foo = myConstants.foo;
  return (<Spinner />);
}
```    

Some test cases:

1. Putting the cursor on `myConstants` and pressing F12 correctly opens `constants/my-constants.js`
2. Putting the cursor on `myConstants`, right-clicking and choosing Go to Type Definition correctly opens `constants/my-constants.js`
3. Putting the cursor on `Spinner` and pressing F12 shows the "No Definition Found" tooltip. **This is unexpected.**
3. Putting the cursor on `Spinner`, right-clicking and choosing Go to Type Definition correctly opens `components/spinner.js`

There is something specific about this project causing this behavior, since F12 seems to work correctly for JSX tags in other projects. F12 also used to work normally for this project in the past. I don't know how to further debug this.

This project does not use TypeScript but I tried adding a `jsconfig.json` file to the project as per #22315, which did not change the behavior above.

**Comment by mjbvz** [comment]
I believe this is a duplicate of https://github.com/Microsoft/TypeScript/issues/15723

Please try setting `"typescript.tsserver.trace": "verbose"` and opening the `TypeScript` section of the output window to confirm this. After running `Go to definition`, you should see a message about `Error processing request`

**Comment by mjbvz** [comment]
As a workaround, you can also try using TypeScript 2.4 in your workspace by installing `typescript@next` and activating this in your workspace https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions

**Comment by patrickgaskill** [comment]
Here's the output:
```
[Trace - 4:49:03 PM] Response received: definition (1410). Request took 3 ms. Success: false . Message: Error processing request. Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at createDefinitionInfo (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:63549:53)
    at Object.getDefinitionAtPosition (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:63440:25)
    at Object.getDefinitionAtPosition (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:71648:38)
    at IOSession.Session.getDefinition (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:76797:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:76338:61)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77649:88
    at IOSession.Session.executeWithRequestId (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77640:28)
    at IOSession.Session.executeCommand (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77649:33)
    at IOSession.Session.onMessage (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77669:35)
    at Interface.<anonymous> (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:78850:27)
[Error - 4:49:03 PM] 'definition' request failed with error.
Error processing request. Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at createDefinitionInfo (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:63549:53)
    at Object.getDefinitionAtPosition (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:63440:25)
    at Object.getDefinitionAtPosition (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:71648:38)
    at IOSession.Session.getDefinition (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:76797:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:76338:61)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77649:88
    at IOSession.Session.executeWithRequestId (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77640:28)
    at IOSession.Session.executeCommand (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77649:33)
    at IOSession.Session.onMessage (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:77669:35)
    at Interface.<anonymous> (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:78850:27)
```

**Comment by mjbvz** [comment]
Thanks for the quick turnaround. Yes, from the error stack this does look like the same issue as Microsoft/TypeScript#15723

We will be picking up TS 2.4 with the fix this month for VSCode 1.14. Until then, you can also try manually upgrading your Typescript version if this issue is getting in the way

Closing as upstream
